### DISCUSSION TITLE
How to open folder upload dialog when clicking dropzone?

### DISCUSSION THREAD

**Discussion by azvast** [react-dropzone/react-dropzone#1157] [Q&A] [answered]
I made folder upload with webkitdirectory attribute in input tag, but I can't make the file open dialog as folder upload dialog.

Here are screenshots of current dialog and wanted dialog.
[https://prnt.sc/YvbeMLLWgBzo]()
[https://prnt.sc/tj_cxztkp8aM](url)

What's the solution for this?

**Comment by rolandjitsu** [comment, marked as answer]
@azvast those 2 URLs don't lead to any relevant info.

As for [webkitdirectory](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/webkitdirectory), that's a non-standard attribute. But, it should be working when the [File System Access API](https://developer.mozilla.org/en-US/docs/Web/API/File_System_Access_API) is not used.

If you're using the latest version of this lib and a modern browser, it will most likely use the FS access API to open the file picker on click. And since we use [showOpenFilePicker](https://developer.mozilla.org/en-US/docs/Web/API/window/showOpenFilePicker), you cannot pick directories. Unfortunately, at the moment, we don't handle using [showDirectoryPicker](https://developer.mozilla.org/en-US/docs/Web/API/window/showDirectoryPicker) when a user sets `webkitdirectory`, but I imagine we could.

Your only option, at the moment, is to disable the FS access API: `useDropzone({useFsAccessApi: false})`. And use the `webkitdirectory` on the input: `<input {...getInputProps({webkitdirectory: "true"})} />`

**Reply by demarchenac** [reply]
But this doesn't accepts files?

**Reply by isdigital-github** [reply]
This works only if you add "true" with quotes to webkitdirectory property. Also, when you select a folder a browser alert will pop up with the following text:
"Upload X files to this site? This will upload all files from "XXXXX". Only do this if you trust the site."

And yes, you can't select individual files then.

**Reply by erictu22** [reply]
Just in case anyone is using the declarative approach (aka not using `useDropdown`), you'll want to specify `useFsAccessApi=false` directly on the `Dropzone` component directly

```
        <Dropzone
          useFsAccessApi={false}
          onDrop={...}
        >
          {({ getRootProps, getInputProps, isDragActive }) => (
            <div {...getRootProps()}>
              <input {...getInputProps({
                dir: uploadType === 'folder' ? "" : undefined, // forces the file dialog to only make folders selectable
                webkitdirectory: uploadType === 'folder' ? "true" : undefined // allows for folder upload
              })} />
...
```

**Comment by azvast** [comment]
Thanks
